chia/wallet/wallet_node_api.py

```diff
@@ -46,7 +46,8 @@ async def new_peak_wallet(self, peak: wallet_protocol.NewPeakWallet, peer: WSChi
         """
         The full node sent as a new peak
         """
-        await self.wallet_node.new_peak_wallet(peak, peer)
+        async with self.wallet_node.new_peak_sem:
```

Comment:
If we are connected to trusted peer (most ofter case now), it makes sense to use a lock. In order to avoid multiple parallel back tracks. Was there any reason you opted for semaphore instead ?

Reply:
change look good oterwise

Reply:
this is just what we do in the node, ill change to a lock, maybe the concurrent sync is redundant in the wallet and just complicates things